BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/beacon/Beacon.java

```diff
@@ -137,23 +139,24 @@ public boolean hasBeenSeenSince(long timestamp) {
         return getLatestAdvertisingPacket().getTimestamp() > timestamp;
     }
 
-    public float getMeanRssi() {
-        return getMeanRssi(3, TimeUnit.SECONDS);
-    }
-
-    public float getMeanRssi(long amount, TimeUnit timeUnit) {
-        List<AdvertisingPacket> recentAdvertisingPackets = (List<AdvertisingPacket>) getAdvertisingPacketsFromLast(amount, timeUnit);
-        int[] recentRssis = AdvertisingPacketUtil.getRssisFromAdvertisingPackets(recentAdvertisingPackets);
-        return AdvertisingPacketUtil.getMeanRssi(recentRssis);
+    public float getRssi(RssiFilter filter) {
+        return filter.filter(advertisingPackets);
     }
 
     public float getDistance() {
         return getDistance(3, TimeUnit.SECONDS);
     }
 
-    public float getDistance(long amount, TimeUnit timeUnit) {
-        float meanRssi = getMeanRssi(amount, timeUnit);
-        return BeaconDistanceCalculator.calculateDistanceTo(this, meanRssi);
+    public float getDistance(long timeWindow, TimeUnit timeUnit) {
+        // TODO evaluate maxSpeed of walking = 1,38889 m/s
```

Comment:
We can't assume that the device (user) is walking. At this level, it is a general purpose library